modules/st2-menu/menu.component.js

```diff
@@ -63,9 +63,35 @@ export default class Menu extends React.Component {
     style: componentStyle,
   }
 
+  componentDidMount () {
+    this.idleLogout();
+  }
+
   docsLink = 'https://docs.stackstorm.com/'
   supportLink = 'https://forum.stackstorm.com/'
 
+  idleLogout() {
+    let t;
+    window.onload = resetTimer;
+    window.onmousemove = resetTimer;
+    window.onmousedown = resetTimer;  // catches touchscreen presses as well      
+    window.ontouchstart = resetTimer; // catches touchscreen swipes as well 
+    window.onclick = resetTimer;      // catches touchpad clicks as well
+    window.onkeydown = resetTimer;   
+    window.addEventListener('scroll', resetTimer, true); 
+
+    function yourFunction() {
+      // your logout code for too long inactivity goes here
+      api.disconnect();
+      window.location.reload();
+    }
+
+    function resetTimer() {
+      window.clearTimeout(t);
+      t = window.setTimeout(yourFunction, 1200000);  // time is in milliseconds application will logout after 20 min. We can set whatever time we want.
```

Comment:
Can we default this to 30 mins and make this a configurable parameter in the application config https://github.com/StackStorm/st2web/blob/master/config.js?